src/Symfony/Component/Validator/Constraints/ContainsValidator.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+class ContainsValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Contains) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\Contains');
+        }
+
+        if (!\is_string($constraint->expected) && !\is_array($constraint->expected) && !$constraint->callback) {
+            throw new ConstraintDefinitionException('Either "expected" or "callback" must be specified on constraint Contains');
+        }
+        if (null === $value) {
+            return;
+        }
+
+        if (!\is_string($value)) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        if ($constraint->callback) {
+            if (!\is_callable($expected = [$this->context->getObject(), $constraint->callback])
+                && !\is_callable($expected = [$this->context->getClassName(), $constraint->callback])
+                && !\is_callable($expected = $constraint->callback)
+            ) {
+                throw new ConstraintDefinitionException('The Contains constraint expects a valid callback');
+            }
+            $expected = $expected();
```

Comment:
Maybe we can inject the value and the context here?

Reply:
and also append some `payload` from `Contains` ?